File: solutions/3090. Maximum Length Substring With Two Occurrences/3090.py
class Solution:
  def maximumLengthSubstring(self, s: str) -> int:
    ans = 0
    count = collections.Counter()

    l = 0
    for r, c in enumerate(s):
      count[c] += 1
      while count[c] > 2:
        count[s[l]] -= 1
        l += 1
      ans = max(ans, r - l + 1)

    return ans

File: solutions/3090. Maximum Length Substring With Two Occurrences/3090.cpp
class Solution {
 public:
  int maximumLengthSubstring(string s) {
    int ans = 0;
    vector<int> count(26);

    for (int l = 0, r = 0; r < s.length(); ++r) {
      ++count[s[r] - 'a'];
      while (count[s[r] - 'a'] > 2)
        --count[s[l++] - 'a'];
      ans = max(ans, r - l + 1);
    }

    return ans;
  }
};
